autonomy/configurations/base.py

```diff
@@ -207,179 +199,146 @@ def validate_config_data(
-
-        if sum(valid) == 4:
-            return True
-
-        raise ValueError("Incorrect number of overrides for count of agents.")
-
     def check_overrides_valid(
         self, overrides: List, env_vars_friendly: bool = False
-    ) -> Dict[ComponentId, Dict[Any, Any]]:
+    ) -> None:
         """Uses the AEA helper libraries to check individual overrides."""
-
-        component_configurations: Dict[ComponentId, Dict] = {}
-        # load the other components.
-
-        for idx, component_configuration_json in enumerate(overrides):
-            component_id, _ = self.process_component_section(
-                idx, component_configuration_json
+        base_validator = validation.ConfigValidator("definitions.json")
+        for component_configuration_json in overrides:
+            configuration, component_id, override_type = self.process_metadata(
+                configuration=copy(component_configuration_json)
             )
-            if component_id in component_configurations:
-                raise ValueError(
-                    f"Configuration of component {component_id} occurs more than once."
+            if override_type == OverrideType.MULTIPLE:
+                for idx in range(self.number_of_agents):
+                    try:
+                        _configuration = cast(Dict, configuration[idx])
+                    except KeyError as e:
+                        raise AEAValidationError(
+                            f"Not enough overrides for component {component_id};"
+                            f" Number of agents: {self.number_of_agents}"
+                        ) from e
+
+                    base_validator.validate_component_configuration(
+                        component_id=component_id,
+                        configuration=_configuration,
+                        env_vars_friendly=env_vars_friendly,
+                    )
+            else:
+                base_validator.validate_component_configuration(
+                    component_id=component_id,
+                    configuration=configuration,
+                    env_vars_friendly=env_vars_friendly,
                 )
-            component_configurations[component_id] = component_configuration_json
 
-        return component_configurations
+    @staticmethod
+    def process_metadata(
+        configuration: Dict,
+    ) -> Tuple[Dict, ComponentId, OverrideType]:
+        """Process component override metadata."""
+
+        component_id = ComponentId(
+            component_type=cast(str, configuration.pop("type")),
+            public_id=PublicId.from_str(
+                cast(
+                    str,
+                    configuration.pop("public_id"),
+                ),
+            ),
+        )
+        _ = configuration.pop("extra", {})
+        overide_type = OverrideType(
+            configuration.pop("override_type", OverrideType.SINGULAR.value)
+        )
```

Comment:
If we don't do this explicitly we this will make the process of validating the overrides much more complicated. I'll try to make the detection for multiple overrides automatic

Reply:
[dc23ba2](https://github.com/valory-xyz/open-autonomy/pull/1576/commits/dc23ba2027fea7592ca886de0db748942e07d9f3)